### DISCUSSION TITLE
Do you use Pluto for Pluto development?

### DISCUSSION THREAD

**Discussion by vochicong** [fonsp/Pluto.jl#1526] [Questions about using Pluto] [answered]
I see that there are several discussions regarding package development using Pluto.
My question is that, do Pluto developers use  Pluto for their development work?
Or, do you recommend using Pluto for development work?

It would be very helpful if someone could show an example workflow of using Pluto for package development!

**Comment by vochicong** [comment, marked as answer]
I found the answer at https://github.com/fonsp/pluto-developer-instructions#setting-up-for-the-first-time.
The developers use VS Code, and Pluto is intended for beginner programmers, not for pros.

**Reply by ctrekker** [reply]
For developing Pluto itself we mostly use VS Code, but for other peripheral packages like PlutoUI we will often use Pluto itself. For example the Table of Contents component (`PlutoUI.TableOfContents`) is built with [this Pluto notebook](https://github.com/fonsp/PlutoUI.jl/blob/main/src/TableOfContents.jl) inside the PlutoUI package.

**Reply by vochicong** [reply]
Have you ever considered developing Pluto using itself? Could you please list up some limitations of Pluto as an IDE for package development?

**Reply by Pangoraw** [reply]
There is already a part of Pluto that is written using Pluto itself ([Firebasey.jl](https://github.com/fonsp/Pluto.jl/blob/ad586710c328a670a8b21a762f304ab06e26f23e/src/webserver/Firebasey.jl) if you want to take a look).

**Reply by vochicong** [reply]
Thank you. Sure, I will.
